src/utils/Subscription.ts

```diff
@@ -57,6 +61,7 @@ function createListenerCollection() {
         isSubscribed = false
 
         if (listener.next) {
+          //@ts-expect-error -- listener.next is always null
```

Comment:
Because `listener` is initialized on line 47 with `next: null`, Typescript interprets it as the following structure:

![image](https://user-images.githubusercontent.com/2777107/123510910-0d25d400-d676-11eb-8053-15368e136412.png)


However when it gets to this line, it causes an error for potentially accessing this `null` field. It doesn't seem to realise that `listener.next` would prevent that from getting accessed.

![image](https://user-images.githubusercontent.com/2777107/123510882-e5367080-d675-11eb-8a12-5e68efa484c3.png)

I'm not entirely sure what's going on here -- either Typescript is wrong that `listener.next` is always `null` (though not sure how that would be), or Typescript is wrong that this `null` check isn't sufficient. Is there another option?

Reply:
I sorted this by explicitly marking the variable as type `Listener` -- that ensured Typescript correctly interprets it as type `Listener | null` rather than `null`.